### ISSUE TITLE
Unable to `expect` the `missing_panics_doc` and `missing_errors_doc` on main function

### ISSUE THREAD

**Issue by CardboardTurkey** [rust-lang/rust-clippy#14491]
### Summary

My code base denies `clippy::missing_panics_doc`, `clippy::missing_errors_doc` and `clippy::allow_attributes` globally. However I also have a pub main function that doesn't need documentation.

If I use `expect` to allow the first two on `main`, clippy complains that the lint is unfulfilled - even though it has stopped complaining about the missing docs. 

Everything works fine on functions that _aren't_ `main`.

Playground: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=acbdff6f05c01e67eee4ad557ff6e06a

### Reproducer

I tried this code:

```rust
#![deny(clippy::missing_panics_doc)]
#![deny(clippy::missing_errors_doc)]
#![deny(clippy::allow_attributes)]

use std::env::var;

#[expect(clippy::missing_panics_doc, reason = "main doesn't need docs")]
#[expect(clippy::missing_errors_doc, reason = "main doesn't need docs")]
pub fn main() -> Result<(), &'static str> {
    let val = var("Hello").unwrap();
    match val.as_str() {
        "hello" => Ok(()),
        _ => Err("Oh no"),
    }
}
```

I expected to see this happen:

No complaints from clippy

Instead, this happened:

Clippy complains about unfulfilled lints

```
    Checking playground v0.0.1 (/playground)
warning: this lint expectation is unfulfilled
 --> src/main.rs:6:10
  |
6 | #[expect(clippy::missing_panics_doc, reason = "main doesn't need docs")]
  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: main doesn't need docs
  = note: `#[warn(unfulfilled_lint_expectations)]` on by default

warning: this lint expectation is unfulfilled
 --> src/main.rs:7:10
  |
7 | #[expect(clippy::missing_errors_doc, reason = "main doesn't need docs")]
  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: main doesn't need docs

warning: `playground` (bin "playground") generated 2 warnings
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.51s
```

### Version

```text
rustc 1.85.1 (4eb161250 2025-03-15)
binary: rustc
commit-hash: 4eb161250e340c8f48f66e2b929ef4a5bed7c181
commit-date: 2025-03-15
host: x86_64-unknown-linux-gnu
release: 1.85.1
LLVM version: 19.1.7
```

**Comment by longfangsong** [comment]
In fact `clippy::missing_panics_doc` and `clippy::missing_errors_doc` is not checked on `main`.

I'm not very sure it is something by design (the document does not clarify this) but I opened #14505 anyway. Let's see how the maitainers think about this.

**Comment by CardboardTurkey** [comment]
@longfangsong Ah you made me realise something. If I remove the `expect`s from the playground example, you're right - clippy doesn't complain about lack of docs on `main`.

However in my actual codebase I'm not using a `main.rs` file, but instead a file in `src/bin`. There it *does* complain about the lack of docs on `main`.
